drivers/timer/stm32_lptim_timer.c

```diff
@@ -61,11 +65,41 @@ static const struct device *const clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_N
  *    0xFFFF / (LSE freq (32768Hz) / 128)
  */
 
-static uint32_t lptim_clock_freq = KHZ(32);
-static int32_t lptim_time_base;
+#if (CONFIG_STM32_LPTIM_CLOCK_LSI)
+static uint32_t lptim_clock_freq = KHZ(32); /* Force STM32_SRC_LSI */
 
-/* The prescaler given by the DTS and to apply to the lptim_clock_freq */
-#define LPTIM_CLOCK_RATIO DT_PROP(DT_DRV_INST(0), st_prescaler)
+/* Kconfig defines the clock source as LSI : check coherency with DTS */
+#if (DT_INST_CLOCKS_CELL_BY_IDX(0, 1, bus) != STM32_SRC_LSI)
+#warning CONFIG_STM32_LPTIM_CLOCK_LSI requires STM32_SRC_LSI (DTS)
+#endif /* STM32_SRC_LSI */
+
+/* Kconfig should set the adapted CONFIG_SYS_CLOCK_TICKS_PER_SEC */
+#if !defined(CONFIG_STM32_LPTIM_TICK_FREQ_RATIO_OVERRIDE)
```

Comment:
Is there a real interest in `CONFIG_STM32_LPTIM_TICK_FREQ_RATIO_OVERRIDE`, now that we understand all constraints related to LPTIM configuration? 
```
	  For LPTIM configuration, a specific tick freq is advised
	  depending on LPTIM input clock:
	  - LSI(32KHz): 4000 ticks/sec
	  - LSE(32678): 4096 ticks/sec
	  To prevent misconfigurations, a dedicated check is implemented
	  in the driver.
	  This options allows to override this check
```
If we need to keep it (which seems to be the case for ZTEST), we might want to update the definition

Reply:
I'm also wondering what is the benefit of all these checks since a default SYS_CLOCK_TICKS_PER_SEC (that verifies the conditions) is already provided.

Reply:
Even if the SYS_CLOCK_TICKS_PER_SEC is given by the soc/arm/st_stm32/common/Kconfig.defconfig.series
an application can still set it in its prj.conf